### PR TITLE
Updating Firefox content

### PR DESCRIPTION
**Pull request by cyarbrough76**
Numerous updates for Firefox content: Guide content xml, .yml, .sh and .var files, Overlay file, Profile and content in macros-bash.jinja.

#### Description:

- _Updated content for checks and remediations for Firefox_

#### Rationale:

- _Address current STIG requirements for Firefox in RHEL_

### CODE REVIEW

**Comment by mildas** [comment]
**Changes identified:**
Rule firefox:
 Found change in bash remediation.
 The rule doesn't occur in any profile nor product.
 Template usage changed in ansible remediation.
 Bash remediation is newly added.
 Templatization usage changed.
Rule firefox_preferences-autofill_passwords:
 Text changed in OVAL check.
Rule firefox_preferences-background_data:
 OVAL check is newly added.
Rule firefox_preferences-development_tools:
 OVAL check is newly added.
Rule firefox_preferences-install_extensions:
 OVAL check is newly added.
Rule firefox_preferences-lock_settings_config_file:
 Text changed in OVAL check.
Rule firefox_preferences-lock_settings_obscure:
 Text changed in OVAL check.
Rule firefox_preferences-ssl_protocol_tls:
 New node inserted to OVAL check.
 Text changed in OVAL check.
 Attribute value changed in OVAL check.
 Node moved within OVAL check.
Rule installed_firefox_version_supported:
 Text changed in OVAL check.
Profile stig on firefox:
 Rule firefox_preferences-javascript_status_bar_changes, firefox_preferences-auto-update_of_firefox, firefox_preferences-javascript_status_bar_text, firefox_preferences-non-secure_page_warning, firefox_preferences-home_page, firefox_preferences-ssl_version_3, firefox_preferences-cookies_user_notice, firefox_preferences-cookies_clear, firefox_preferences-ssl_version_2 removed from stig profile.
 Rule firefox_preferences-development_tools, firefox_preferences-install_extensions, firefox_preferences-background_data added to stig profile.
Macro bash_coredump_config_set:
 In Bash remediation for coredump_disable_storage.
 In Bash remediation for coredump_disable_backtraces.
Macro bash_firefox_js_setting:
 In Bash remediation for firefox_preferences-lock_settings_obscure.
 In Bash remediation for firefox_preferences-lock_settings_config_file.
Macro bash_firefox_cfg_setting:
 In Bash remediation for firefox_preferences-background_data.
 In Bash remediation for firefox_preferences-autofill_passwords.
 In Bash remediation for firefox_preferences-home_page.
 In Bash remediation for firefox_preferences-non-secure_page_warning.
 In Bash remediation for firefox_preferences-ssl_version_3.
 In Bash remediation for firefox_preferences-ssl_protocol_tls.
 In Bash remediation for firefox_preferences-cookies_user_notice.
 In Bash remediation for firefox_preferences-addons_plugin_updates.
 In Bash remediation for firefox_preferences-javascript_window_changes.
 In Bash remediation for firefox_preferences-javascript_status_bar_changes.
 In Bash remediation for firefox_preferences-ssl_version_2.
 In Bash remediation for firefox_preferences-pop-up_windows.
 In Bash remediation for firefox_preferences-install_extensions.
 In Bash remediation for firefox_preferences-autofill_forms.
 In Bash remediation for firefox_preferences-development_tools.
 In Bash remediation for firefox_preferences-auto-update_of_firefox.
 In Bash remediation for firefox_preferences-auto-download_actions.
 In Bash remediation for firefox_preferences-javascript_window_resizing.
 In Bash remediation for firefox_preferences-verification.
 In Bash remediation for firefox_preferences-open_confirmation.
 In Bash remediation for firefox_preferences-search_update.
 In Bash remediation for firefox_preferences-cookies_clear.
 In Bash remediation for firefox_preferences-javascript_status_bar_text.
 In Bash remediation for firefox_preferences-javascript_context_menus.
 In Bash remediation for firefox_preferences-shell_protocol.
 In Bash remediation for firefox_preferences-password_store.

**Recommended tests to execute:**
 build_product firefox
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-firefox-ds.xml firefox_preferences-autofill_passwords
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-firefox-ds.xml firefox_preferences-background_data
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-firefox-ds.xml firefox_preferences-development_tools
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-firefox-ds.xml firefox_preferences-install_extensions
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-firefox-ds.xml firefox_preferences-lock_settings_config_file
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-firefox-ds.xml firefox_preferences-lock_settings_obscure
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-firefox-ds.xml firefox_preferences-ssl_protocol_tls
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-firefox-ds.xml installed_firefox_version_supported
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-firefox-ds.xml stig

File: firefox/guide/firefox/firefox_preferences-background_data/oval/firefox.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<def-group>
+  <definition class="compliance" id="firefox_preferences-background_data"  version="1">
+    <metadata>
+      <title>Background submission of information to Mozilla must be disabled.</title>
```
Comment (redhatrises) [inline firefox/guide/firefox/firefox_preferences-background_data/oval/firefox.xml:4]: Let's bring this inline with the other rules. Please make the title:
`Disable Background Information Submission`

File: firefox/guide/firefox/firefox_preferences-background_data/oval/firefox.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      <affected family="unix">
+        <platform>Mozilla Firefox</platform>
+      </affected>
+      <description>There should be no background submission of technical and other information 
+	  from DoD computers to Mozilla with portions posted publically.</description>
```
Comment (redhatrises) [inline firefox/guide/firefox/firefox_preferences-background_data/oval/firefox.xml:9]: Let's make this usable for more people. Take out `from DoD computers`

File: firefox/guide/firefox/firefox_preferences-development_tools/rule.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+documentation_complete: true
+
+title: 'Firefox Development Tools Must Be Disabled.'
```
Comment (redhatrises) [inline firefox/guide/firefox/firefox_preferences-development_tools/rule.yml:3]: Let's make this: `Disable Firefox Development Tools`

File: shared/macros-bash.jinja [hunk 4]
```diff
@@ -1,20 +1,5 @@
...
-echo "Setting=$varname1" >> config_file
-
-#}}
-{{%- macro bash_instantiate_variables() -%}}
-{{%- for name in varargs -%}}
-populate {{{ name }}}
-{{# this line is intentionally left blank #}}
-{{% endfor -%}}
-{{%- endmacro -%}}
-
```
Comment (redhatrises) [inline shared/macros-bash.jinja:17]: @cyarbrough76 why this change?

File: shared/macros-bash.jinja [hunk 5]
```diff
@@ -86,21 +71,6 @@ populate {{{ name }}}
...
-        create=true,
-        insert_after="",
-        insert_before="",
-        insensitive=true,
-        separator="=",
-        separator_regex="=",
-        prefix_regex="^")
-    }}}
-{{%- endmacro -%}}
-
```
Comment (redhatrises) [inline shared/macros-bash.jinja:103]: @cyarbrough76 why this change?

File: firefox/guide/firefox/firefox_preferences-ssl_protocol_tls/rule.yml [hunk 6]
```diff
@@ -2,7 +2,8 @@ documentation_complete: true
 
 title: 'Enable TLS Usage in Firefox'
 
-description: 'To enable TLS, set <tt>security.enable_tls</tt> to <tt>true</tt>.'
+description: 'To enable TLS, set <tt>security.tls.version.min</tt> to <tt>2</tt>
+ and set <tt>security.tls.version.max</tt> to <tt>4</tt>.'
```
Comment (redhatrises) [inline firefox/guide/firefox/firefox_preferences-ssl_protocol_tls/rule.yml:6]: This should really be a new rule since it is changing 2 separate configuration items.

File: shared/macros-bash.jinja [hunk 7]
```diff
@@ -361,8 +331,10 @@ for firefox_dir in ${firefox_dirs}; do
     if [ -d "${firefox_dir}" ]; then
         # Make sure the Firefox .cfg file exists and has the appropriate permissions
         if ! [ -f "${firefox_dir}/${firefox_cfg}" ] ; then
-            touch "${firefox_dir}/${firefox_cfg}"
+            echo "//" "${firefox_dir}/${firefox_cfg}"
```
Comment (redhatrises) [inline shared/macros-bash.jinja:334]: @cyarbrough76 why this change?

File: firefox/guide/firefox/firefox_preferences-background_data/rule.yml [hunk 8]
```diff
@@ -0,0 +1,29 @@
+documentation_complete: true
+
+title: 'Background submission of information to Mozilla must be disabled.'
```
Comment (redhatrises) [inline firefox/guide/firefox/firefox_preferences-background_data/rule.yml:3]: Please make the title:
`Disable Background Information Submission`

File: firefox/guide/firefox/firefox_preferences-development_tools/oval/firefox.xml [hunk 9]
```diff
@@ -0,0 +1,24 @@
+<def-group>
+  <definition class="compliance" id="firefox_preferences-development_tools"  version="1">
+    <metadata>
+      <title>Firefox Development Tools Must Be Disabled.</title>
```
Comment (redhatrises) [inline firefox/guide/firefox/firefox_preferences-development_tools/oval/firefox.xml:4]: Please make the title:
`Disable Background Information Submission`

File: firefox/guide/firefox/firefox_preferences-install_extensions/oval/firefox.xml [hunk 10]
```diff
@@ -0,0 +1,24 @@
+<def-group>
+  <definition class="compliance" id="firefox_preferences-install_extensions"  version="1">
+    <metadata>
+      <title>Extensions install must be disabled.</title>
```
Comment (redhatrises) [inline firefox/guide/firefox/firefox_preferences-install_extensions/oval/firefox.xml:4]: Please change title to `Disable Extension Installation`

File: firefox/guide/firefox/firefox_preferences-install_extensions/rule.yml [hunk 11]
```diff
@@ -0,0 +1,29 @@
+documentation_complete: true
+
+title: 'Extensions install must be disabled.'
```
Comment (redhatrises) [inline firefox/guide/firefox/firefox_preferences-install_extensions/rule.yml:3]: Please change title to `Disable Extension Installation`

File: shared/macros-bash.jinja (hunk 4 above)
Comment (cyarbrough76) [reply shared/macros-bash.jinja:17]: Unintentional. working

File: shared/macros-bash.jinja (hunk 5 above)
Comment (cyarbrough76) [reply shared/macros-bash.jinja:103]: Unintentional. Working on it

**Comment by redhatrises** [comment]
@cyarbrough76 Thanks for doing this very important update that many contributors, users, and others have been asking for!

**Comment by mildas** [comment]
**Changes identified:**
Rule firefox:
 Bash remediation is newly added.
 The rule doesn't occur in any profile nor product.
 Template usage changed in ansible remediation.
 Templatization usage changed.
 Found change in bash remediation.
Rule firefox_preferences-autofill_passwords:
 Text changed in OVAL check.
Rule firefox_preferences-background_data:
 OVAL check is newly added.
Rule firefox_preferences-development_tools:
 OVAL check is newly added.
Rule firefox_preferences-install_extensions:
 OVAL check is newly added.
Rule firefox_preferences-lock_settings_config_file:
 Text changed in OVAL check.
Rule firefox_preferences-lock_settings_obscure:
 Text changed in OVAL check.
Rule firefox_preferences-ssl_protocol_tls:
 Text changed in OVAL check.
 Node moved within OVAL check.
 New node inserted to OVAL check.
 Attribute value changed in OVAL check.
Rule installed_firefox_version_supported:
 Text changed in OVAL check.
Profile stig on firefox:
 Rule firefox_preferences-background_data, firefox_preferences-development_tools, firefox_preferences-install_extensions added to stig profile.
 Rule firefox_preferences-home_page, firefox_preferences-cookies_user_notice, firefox_preferences-auto-update_of_firefox, firefox_preferences-javascript_status_bar_text, firefox_preferences-ssl_version_2, firefox_preferences-ssl_version_3, firefox_preferences-cookies_clear, firefox_preferences-non-secure_page_warning, firefox_preferences-javascript_status_bar_changes removed from stig profile.
Macro bash_firefox_js_setting:
 In Bash remediation for firefox_preferences-lock_settings_obscure.
 In Bash remediation for firefox_preferences-lock_settings_config_file.
Macro bash_firefox_cfg_setting:
 In Bash remediation for firefox_preferences-ssl_version_3.
 In Bash remediation for firefox_preferences-autofill_passwords.
 In Bash remediation for firefox_preferences-autofill_forms.
 In Bash remediation for firefox_preferences-ssl_version_2.
 In Bash remediation for firefox_preferences-password_store.
 In Bash remediation for firefox_preferences-cookies_clear.
 In Bash remediation for firefox_preferences-open_confirmation.
 In Bash remediation for firefox_preferences-javascript_status_bar_text.
 In Bash remediation for firefox_preferences-development_tools.
 In Bash remediation for firefox_preferences-verification.
 In Bash remediation for firefox_preferences-background_data.
 In Bash remediation for firefox_preferences-auto-download_actions.
 In Bash remediation for firefox_preferences-pop-up_windows.
 In Bash remediation for firefox_preferences-shell_protocol.
 In Bash remediation for firefox_preferences-javascript_context_menus.
 In Bash remediation for firefox_preferences-auto-update_of_firefox.
 In Bash remediation for firefox_preferences-javascript_status_bar_changes.
 In Bash remediation for firefox_preferences-addons_plugin_updates.
 In Bash remediation for firefox_preferences-cookies_user_notice.
 In Bash remediation for firefox_preferences-search_update.
 In Bash remediation for firefox_preferences-ssl_protocol_tls.
 In Bash remediation for firefox_preferences-install_extensions.
 In Bash remediation for firefox_preferences-javascript_window_resizing.
 In Bash remediation for firefox_preferences-home_page.
 In Bash remediation for firefox_preferences-non-secure_page_warning.
 In Bash remediation for firefox_preferences-javascript_window_changes.

**Recommended tests to execute:**
 build_product firefox
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-firefox-ds.xml firefox_preferences-autofill_passwords
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-firefox-ds.xml firefox_preferences-background_data
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-firefox-ds.xml firefox_preferences-development_tools
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-firefox-ds.xml firefox_preferences-install_extensions
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-firefox-ds.xml firefox_preferences-lock_settings_config_file
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-firefox-ds.xml firefox_preferences-lock_settings_obscure
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-firefox-ds.xml firefox_preferences-ssl_protocol_tls
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-firefox-ds.xml installed_firefox_version_supported
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-firefox-ds.xml stig

File: firefox/guide/firefox/firefox_preferences-background_data/oval/firefox.xml (hunk 1 above)
Comment (cyarbrough76) [reply firefox/guide/firefox/firefox_preferences-background_data/oval/firefox.xml]: Edited the title accordingly

File: firefox/guide/firefox/firefox_preferences-background_data/oval/firefox.xml (hunk 2 above)
Comment (cyarbrough76) [reply firefox/guide/firefox/firefox_preferences-background_data/oval/firefox.xml]: removed

File: firefox/guide/firefox/firefox_preferences-development_tools/rule.yml (hunk 3 above)
Comment (cyarbrough76) [reply firefox/guide/firefox/firefox_preferences-development_tools/rule.yml]: Edited the title accordingly

File: shared/macros-bash.jinja (hunk 4 above)
Comment (cyarbrough76) [reply shared/macros-bash.jinja]: Replaced the function.

File: shared/macros-bash.jinja (hunk 7 above)
Comment (thenefield) [reply shared/macros-bash.jinja:364]: These changes facilitate aligning the remediation with the latest DISA STIG and benchmark.

While ideally, the file name really shouldn't matter, but since that's what the DISA benchmark specifically looks for. It's easier just to pass a scan than explain for each failure how it's a pass.

The changes for adding a '//' in the first line of the file is because the first line of the file is ignored. This is also outlined in the STIG.

The file path changes were made to give the default preference the location the STIG looks for.

The variables at the top also needed a slash to properly resolve.

File: shared/macros-bash.jinja (hunk 5 above)
Comment (cyarbrough76) [reply shared/macros-bash.jinja]: Replaced the function

File: firefox/guide/firefox/firefox_preferences-background_data/rule.yml (hunk 8 above)
Comment (cyarbrough76) [reply firefox/guide/firefox/firefox_preferences-background_data/rule.yml]: Edited the title accordingly

File: firefox/guide/firefox/firefox_preferences-development_tools/oval/firefox.xml (hunk 9 above)
Comment (cyarbrough76) [reply firefox/guide/firefox/firefox_preferences-development_tools/oval/firefox.xml]: Edited the title accordingly

File: firefox/guide/firefox/firefox_preferences-install_extensions/oval/firefox.xml (hunk 10 above)
Comment (cyarbrough76) [reply firefox/guide/firefox/firefox_preferences-install_extensions/oval/firefox.xml]: Edited the title accordingly

File: firefox/guide/firefox/firefox_preferences-install_extensions/rule.yml (hunk 11 above)
Comment (cyarbrough76) [reply firefox/guide/firefox/firefox_preferences-install_extensions/rule.yml]: Edited the title accordingly

File: firefox/guide/firefox/firefox_preferences-ssl_protocol_tls/rule.yml (hunk 6 above)
Comment (thenefield) [reply firefox/guide/firefox/firefox_preferences-ssl_protocol_tls/rule.yml:6]: The following STIG requirement requires both values:

https://www.stigviewer.com/stig/mozilla_firefox/2019-12-12/finding/V-15983

We need to map a single rule to each requirement.

Comment (redhatrises) [reply firefox/guide/firefox/firefox_preferences-ssl_protocol_tls/rule.yml:6]: > This should really be a new rule since it is changing 2 separate configuration items.

Understood. We need the rule to be broken out into 2 to meet the CCI.

**Comment by redhatrises** [comment]
There's more changes that need to take place to bring this in line with some of the configuration changes in RHEL, but not going to hold that up for this great progress. Thanks!
